include/openmc/plot.h

```diff
@@ -66,24 +89,95 @@ enum class PlotBasis {
 };
 
 enum class PlotColorBy {
-  cells = 1,
-  mats = 2
+  cells = 0,
+  mats = 1
 };
 
 //===============================================================================
 // Plot class
 //===============================================================================
-struct PlotBase {
+class PlotBase {
+public:
+  template<class T> T get_map() const;
+
   // Members
+public:
   Position origin_; //!< Plot origin in geometry
   Position width_; //!< Plot width in geometry
   PlotBasis basis_; //!< Plot basis (XY/XZ/YZ)
-  std::array<int, 3> pixels_; //!< Plot size in pixels
+  std::array<size_t, 3> pixels_; //!< Plot size in pixels
   int level_; //!< Plot universe level
 };
 
-class Plot : public PlotBase
-{
+template<class T>
+T PlotBase::get_map() const {
+
+  size_t width = pixels_[0];
+  size_t height = pixels_[1];
+
+  // get pixel size
+  double in_pixel = (width_[0])/static_cast<double>(width);
+  double out_pixel = (width_[1])/static_cast<double>(height);
+
+  // size data array
+  T data(width, height);
+
+  // setup basis indices and initial position centered on pixel
+  int in_i, out_i;
+  Position xyz = origin_;
+  switch(basis_) {
+  case PlotBasis::xy :
+    in_i = 0;
+    out_i = 1;
+    break;
+  case PlotBasis::xz :
+    in_i = 0;
+    out_i = 2;
+    break;
+  case PlotBasis::yz :
+    in_i = 1;
+    out_i = 2;
+    break;
+  }
+
+  // set initial position
+  xyz[in_i] = origin_[in_i] - width_[0] / 2. + in_pixel / 2.;
+  xyz[out_i] = origin_[out_i] + width_[1] / 2. - out_pixel / 2.;
+
+  // arbitrary direction
+  Direction dir = {0.5, 0.5, 0.5};
```

Comment:
Wasn't this direction problematic for you?

Reply:
Ah! It was yeah, must've missed that in a rebase. Thanks!

Reply:
Or a yet-to-occur rebase... looks like that's part of the current conflict with develop.